### DISCUSSION TITLE
Best module for voice?

### DISCUSSION THREAD

**Discussion by gradasso-lesperance** [RoEdAl/asterisk-chan-quectel#22] [Q&A] [answered]
(had this in Issues because I am an idiot who forgot Github Discussions exist)

Hi there,

I have been unsuccessful utilising one of my Huawei modems (using chan_dongle) for voice calling using latest Asterisk/FreePBX.

It works well for SMS messages, however.

Now looking for another piece of hardware (open to any suggestions) to get a decent call quality from GSM using Asterisk.

I have looked around and this place seems like the most up-to-date group of people on the subject.

Or should I go only with a "proper" GSM Gateway hardware if I want to get good audio call quality? (costing $200+)

Would appreciate any wise and experienced hints!

**Comment by gradasso-lesperance** [comment]
@RoEdAl would you be able to advise on this please? I know you're really busy, but value your opinion :)

**Comment by RoEdAl** [comment]
Sound quality is a tricky part. Even on "professional" GSM gateways sound quality may be really bad. 

Here are my hints:

* Use solid (quiet) power adapter.
    
    Solid power adapter is a must. It does not have to be expensive. Use adapters from manufacturers you know.

* Prefer Mini PCI-E to USB adapters with additional power socket.

    USB2 is not able to provide enough power to LTE modules. On many boards USB3 ports have serious powering issues.

* Avoid Mini PCI-E to USB adapters with powering issues.

    Adapters with wrongly designed power paths may decrease radio signal quality or/and make module unstable.

* Prefer computers/boards with Mini PCI-E ports.
* Use solid antennas.
  
    There are many antennas on the market with **negative** gain. Yes. Many cheap antennas work like attenuators! Again. Use antennas from manufacturers you trust.

**Reply by gradasso-lesperance** [reply]
@RoEdAl wow! what an awesome response! what about the modules - which ones are the best in your opinion?

**Reply by RoEdAl** [reply, marked as answer]
There are only two "officially" supported devices: *Quectel EC25-E* and *SimCOM SIM7600G-H*. I prefer the former one for two reasons:

* VoLTE - I am unable to get VoLTE on *SimCOM* module. But other people report fully functional VoLTE on their *SimCOM* modules.
* Dedicated pseudo audio device on Quectel - this seems to me a better approach of voice handling than voice over serial port.

**Reply by gradasso-lesperance** [reply]
@RoEdAl ok thanks a lot!
